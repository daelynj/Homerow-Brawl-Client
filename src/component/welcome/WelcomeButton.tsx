import * as React from "react";
interface Props {
  setRoomID: (newRoomID: number) => void;
  setIsLoaded: (newIsLoaded: boolean) => void;
  setError: (newError: any) => void;
}

export const WelcomeButton = (props: Props) => {
  const handleEvent = () => {
    fetch("http://localhost:3000/api/rooms", {
      method: "POST",
      mode: "cors"
    })
      .then(response => response.json())
      .then(
        result => {
          props.setRoomID(result.id);
          props.setIsLoaded(true);
        },
        error => {
          props.setIsLoaded(true);
          props.setError(error);
        }
      );
  };

  return (
    <div>
      <button onClick={handleEvent}>Generate lobby</button>
    </div>
  );
};
