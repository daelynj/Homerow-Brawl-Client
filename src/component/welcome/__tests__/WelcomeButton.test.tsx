import * as React from "react";
import { WelcomeButton } from "../WelcomeButton";
import renderer from "react-test-renderer";
import { mount, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("WelcomeButton", () => {
  describe("The button render", () => {
    const mockHandleEvent = jest.fn();

    it("renders a Generate lobby button", () => {
      const tree = renderer
        .create(<WelcomeButton handleEvent={mockHandleEvent} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe("onClick", () => {
    const mockHandleEvent = jest.fn();

    it("triggers an event", () => {
      const wrapper = mount(<WelcomeButton handleEvent={mockHandleEvent} />);

      wrapper.find("button").simulate("click");
      expect(mockHandleEvent).toHaveBeenCalled();
    });
  });
});
