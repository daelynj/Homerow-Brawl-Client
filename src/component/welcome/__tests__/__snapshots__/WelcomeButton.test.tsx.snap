// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeButton The button render renders a Generate lobby button 1`] = `
<button
  onClick={[MockFunction]}
>
  Generate lobby
</button>
`;
